scripts/bash/run_enceladus.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+
+# Copyright 2018-2019 ABSA Group Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Command line for the script itself
+
+# Show spark-submit command line without actually running it (--dry-run)
+DRY_RUN=""
+
+# Command line defaults for 'spark-submit'
+MASTER="yarn"
+DEPLOY_MODE="client"
+NUM_EXECUTORS=""
+EXECUTOR_MEMORY="4G"
+DRIVER_CORES="4"
+DRIVER_MEMORY="8G"
+
+# Command like default for the job
+DATASET_NAME=""
+DATASET_VERSION=""
+REPORT_DATE=""
+REPORT_VERSION=""
+RAW_FORMAT=""
+ROWTAG=""
+DELIMITER=""
+HEADER=""
+TRIM_VALUES=""
+MAPPING_TABLE_PATTERN="''"
+FOLDER_PREFIX=""
+
+# Security command line defaults
+MENAS_CREDENTIALS_FILE=""
+MENAS_AUTH_KEYTAB=""
+
+# Parse command line (based on https://stackoverflow.com/questions/192249/how-do-i-parse-command-line-arguments-in-bash)
+POSITIONAL=()
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --dry-run)
+    DRY_RUN="1"
+    shift # past argument
+    ;;
+    --num-executors)
+    NUM_EXECUTORS="$2"
+    shift 2 # past argument and value
+    ;;
+    --executor-memory)
+    EXECUTOR_MEMORY="$2"
+    shift 2 # past argument and value
+    ;;
+    --master)
+    MASTER="$2"
+    shift 2 # past argument and value
+    ;;
+    --deploy-mode)
+    DEPLOY_MODE="$2"
+    shift 2 # past argument and value
+    ;;
+    --driver-cores)
+    DRIVER_CORES="$2"
+    shift 2 # past argument and value
+    ;;
+    --driver-memory)
+    DRIVER_MEMORY="$2"
+    shift 2 # past argument and value
+    ;;
+    --jar)
+    JAR="$2"
+    shift 2 # past argument and value
+    ;;
+    --class)
+    JAR="$2"
+    shift 2 # past argument and value
+    ;;
+    -D|--dataset-name)
+    DATASET_NAME="$2"
+    shift 2 # past argument and value
+    ;;
+    -d|--dataset-version)
+    DATASET_VERSION="$2"
+    shift 2 # past argument and value
+    ;;
+    -R|--report-date)
+    REPORT_DATE="$2"
+    shift 2 # past argument and value
+    ;;
+    -r|--report-version)
+    REPORT_VERSION="$2"
+    shift 2 # past argument and value
+    ;;
+    --folder-prefix)
+    FOLDER_PREFIX="$2"
+    shift 2 # past argument and value
+    ;;
+
+    -f|--raw-format)
+    RAW_FORMAT="$2"
+    shift 2 # past argument and value
+    ;;  
+    --rowtag)
+    ROWTAG="$2"
+    shift 2 # past argument and value
+    ;;
+    --delimiter)
+    DELIMITER="\\$2"
```

Comment:
Why is this escaped like this?

Reply:
If a user specifies a character that needs escaping, like `\|`, the escape character gets consumed when reassigning variables. So I've added '\\' to always escape.